### PR TITLE
Refactor hackathon application service

### PR DESCRIPTION
**Pull request by Alexpud**
Made a refactoring as to move the method calls to HackathonService in the Domain to go through the ApplicationLayer, also the conversion from entities to DTOs and DTOs to entities is made on the ApplicationLayer utilizing AutoMapper. I did this in steps to avoid another huge PR.

### CODE REVIEW

File: RemoteHackathon.Application/Dtos/HackathonDTO.cs [hunk 1]
```diff
@@ -0,0 +1,28 @@
+using System;
```
Comment (roymoran) [inline RemoteHackathon.Application/Dtos/HackathonDTO.cs]: There are type conflicts with swagger, could you take a look at this?
Screen Shot 2019-09-23 at 1 21 32 AM

Comment (roymoran) [inline RemoteHackathon.Application/Dtos/HackathonDTO.cs]: Why do we now have 2 DTO directories? It seems to conflict with swagger.

Comment (Alexpud) [reply RemoteHackathon.Application/Dtos/HackathonDTO.cs]: Well, one is for the application services DTOs the other ones is in the Domain. Is it generating conflict? We could give it another name so it doesn't happen

Comment (Alexpud) [reply RemoteHackathon.Application/Dtos/HackathonDTO.cs]: Also, where did you find this? Was on azure or running something locally?

Comment (roymoran) [reply RemoteHackathon.Application/Dtos/HackathonDTO.cs]: @Alexpud this output was from the console - running the application in the terminal.

Comment (roymoran) [reply RemoteHackathon.Application/Dtos/HackathonDTO.cs]: @Alexpud ok - yeah maybe different naming will be a good idea.

**Pull request merged**